consensus/misc/eip1559/eip1559.go

```diff
@@ -93,8 +130,12 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header, time uint64)
 	if config.IsHolocene(time) {
```

Comment:
See spec comment. It should check the hardfork time of the `parent` header if it's going to make assumptions on the `parent` header contents.
On the first block of the hardfork, with a pre-holocene parent-header, we could default to the old Canyon constants.

Reply:
I optimistically approved the PR, but this should be addressed before we can merge

Reply:
done